frontend/src/screens/apps/ShowApp.tsx

```diff
@@ -17,6 +17,11 @@ function ShowApp() {
   if (!app || !info) {
     return <Loading />;
   }
+
+  if (app && "error" in app) {
```

Comment:
We should not do this:

- Why navigate to 404 on **any** error?
- By immediately navigating on any error it will make it hard to debug when this happens
- Why not show the error on the page the error occurred?

Reply:
> Why navigate to 404 on any error?

Because they were looking for an app which doesn't exist

> Why not show the error on the page the error occurred?

There won't be any page to show the error right? We can however return to home screen and show an error toast there saying app doesn't exist

Reply:
> Because they were looking for an app which doesn't exist

This is only one possible reason for an error. You would need to check the response code for that, but I wouldn't do any special handling. Maybe we just display the error directly for now.